modules/services/gpg-agent.nix

```diff
@@ -218,6 +218,21 @@ in {
         '';
       };
 
+      noAllowExternalCache = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Tell Pinentry not to enable features which use an external cache for
+          passphrases.
+
+          Some desktop environments prefer to unlock all credentials with one
+          master password and may have installed a Pinentry which employs an
+          additional external cache to implement such a policy. By using this
+          option the Pinentry is advised not to make use of such a cache and
+          instead always ask the user for the requested passphrase.
```

Comment:
I am not much familiar with gpg.
is `allow-external-cache` allowed along with `no-allow-external-cache` ? 

I like the effort to describe the option but I found it a bit long. Can this be summed up by `ask pin not to cache password`. What if the pin package ignores the request ? does it collide with other options like `defaultCacheTtl` in which case some warnings might be of interest ?

Reply:
I took the description directly from gpg-agent, I think that without familiarity with gpg-agent and gpg, it wil likely not make much sense. It cannot be summed up as ask pin not to cache password.

It doesn't collide with other options. I am using it with all the options home manager currently provides.

Reply:
Also, I'm not aware of an `allow-external-cache` option, at least it's not present in any documentation https://www.gnupg.org/documentation/manuals/gnupg/Agent-Options.html.